terraform/01-jenkins-server/main.tf

```diff
@@ -0,0 +1,80 @@
+provider "aws" {
+  profile = "default"
+  region  = local.aws_region
+}
+
+locals {
+  aws_region          = "us-east-1"
+  internal_cidr_block = "10.10.10.0/24"
+  external_cidr_block = "0.0.0.0/0"
+}
+
+resource "aws_vpc" "project_vpc" {
+  cidr_block = local.internal_cidr_block
+}
+
+resource "aws_subnet" "project_subnet" {
+  vpc_id            = aws_vpc.project_vpc.id
+  cidr_block        = local.internal_cidr_block
+  availability_zone = "${local.aws_region}d"
+}
+
+resource "aws_security_group" "project_security_group" {
+  name   = "project_security_group"
+  vpc_id = aws_vpc.project_vpc.id
+
+  ingress {
+    from_port   = 22
+    protocol    = "tcp"
+    to_port     = 22
+    cidr_blocks = [local.external_cidr_block]
+  }
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = [local.external_cidr_block]
+  }
+
+  lifecycle {
+    create_before_destroy = true
```

Comment:
What for? What problem are you solving with this?

Reply:
Deleted